### PR TITLE
Added Android device commands

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
## Change list

Added Android device commands 
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Added Android device commands
```
isKeyboardShown
getDisplayDensity
getSystemBars
```
Added,
```io.appium.java_client.android.IsKeyboardShown``` API and
```io.appium.java_client.HasDeviceDetails``` API.

@TikhomirovSergey Please review

### CODE REVIEW

**Comment by SrinivasanTarget** [comment]
git checkout -b SrinivasanTarget-AndroidDeviceCommands master
git pull https://github.com/SrinivasanTarget/java-client.git AndroidDeviceCommands

**Comment by SrinivasanTarget** [comment]
@imurchie Can you please check this? https://gist.github.com/SrinivasanTarget/d564095d2009779bed69a2d6ef8f0b8e.

I'm running from source with latest android driver and base driver.

**Comment by imurchie** [comment]
@SrinivasanTarget This should be a `GET` request. https://github.com/appium/appium-base-driver/blob/master/lib/mjsonwp/routes.js#L333

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey Have fixed this PR.

**Comment by TikhomirovSergey** [comment]
This PR is in the conflicted state

**Comment by SrinivasanTarget** [comment]
Yup @TikhomirovSergey Will close this and raise a new one probably in sometime.

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey I have rebased this to master and fixed the conflicts now.

File: src/main/java/io/appium/java_client/HasDeviceDetails.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package io.appium.java_client;
+
+import static io.appium.java_client.MobileCommand.GET_DISPLAY_DENSITY;
+import static io.appium.java_client.MobileCommand.GET_SYSTEM_BARS;
+
+import java.util.Map;
+
+public interface HasDeviceDetails extends ExecutesMethod {
```
Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/HasDeviceDetails.java]: It seems that methods that declared by this interface are Android-specific. Please take a look at 
https://github.com/search?q=org%3Aappium+DISPLAY+DENSITY&type=Code

https://github.com/search?q=org%3Aappium+GET+SYSTEM+BARS&type=Code

I think it has more sense to move this interface to the ```io.appium.java_client.android```

File: src/main/java/io/appium/java_client/android/IsKeyboardShown.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package io.appium.java_client.android;
+
+import static io.appium.java_client.android.AndroidMobileCommandHelper.isKeyboardShownCommand;
+
+import io.appium.java_client.CommandExecutionHelper;
+import io.appium.java_client.ExecutesMethod;
+
+public interface IsKeyboardShown extends ExecutesMethod {
```
Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/android/IsKeyboardShown.java]: I think it has more sense to unite this interface with ```HasDeviceDetails```

File: src/test/java/io/appium/java_client/android/AndroidSearchingTest.java [hunk 3]
```diff
@@ -63,4 +65,10 @@ public void setup() throws Exception {
                 + "new UiSelector().text(\"Radio Group\"));");
         assertNotNull(radioGroup.getLocation());
     }
+
+    @Test public void deviceDetailsAndKeyboardTest() {
```
Comment (TikhomirovSergey) [inline src/test/java/io/appium/java_client/android/AndroidSearchingTest.java]: Why this method is here? Maybe it has more sense to move it to ```AndroidDriverTest```

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey Fixed review comments now

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey Ping

**Pull request merged**